File: solutions/0011-container-with-most-water/solution.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        p1 = 0
        p2 = len(height)-1
        max_area = 0
        for i in range(len(height)):
            area = (p2-p1) * min(height[p1],height[p2])
            if area > max_area:
                max_area = area
            if height[p1]<height[p2]:
                p1 = p1+1
            else:
                p2 = p2-1
        return max_area

File: solutions/0011-container-with-most-water/solution.java
class Solution {
    public int maxArea(int[] height) {
        int i = 0;
        int j = height.length-1;
        int w = j;
        int maxx = 0;
        while(i<j){
            maxx = Math.max(maxx,(w*Math.min(height[i],height[j])));
            if (height[i]<height[j]){
                i++;
                w--;
            }
            else{
                j--;
                w--;
            }
            

            
        }
        return maxx;
        
    }
}
